### ISSUE TITLE
The training effect of custom data is poor

### ISSUE THREAD

**Issue by cici19850** [maturk/dn-splatter#78]
Hello, thanks for the code.
After setting up the environment, run the following command：

> python dn_splatter/scripts/convert_colmap.py --image-path data\meetingroom\images --use-gpu

> python dn_splatter/scripts/normals_from_pretrain.py --data-dir data\meetingroom --img-dir-name data\meetingroom\images 

> python dn_splatter/scripts/align_depth.py --data data\meetingroom

> ns-train dn-splatter --pipeline.model.use-depth-loss True --pipeline.model.depth-loss-type PearsonDepth --pipeline.model.depth-lambda 0.2 --pipeline.model.use-normal-loss True --pipeline.model.use-normal-tv-loss True --pipeline.model.normal-supervision mono coolermap --data data\meetingroom --normals-from pretrained --normal-format opencv --load-depths True --load_normals True

The training effect is as follows，Is there any way to improve the effect？
<img width="896" alt="1728544080446" src="https://github.com/user-attachments/assets/e84c326c-31c6-426f-b98f-5bbdcf354ad2">

**Comment by maturk** [comment]
Hey, it is difficult to say what could be a problem. The first thing I notice are the white walls, and these are quite dificult to regularize well with monocular depth regularization. 

Can you do a comparison with only splatfacto and compare with dn_splatter? Does dn_splatter do worse than splatfacto? You can get eval metrics using ns-eval command to see PSNR.

**Comment by cici19850** [comment]
> Hey, it is difficult to say what could be a problem. The first thing I notice are the white walls, and these are quite difficult to regularize well with monocular depth regularization. How many images are in your dataset, and are they clear? (no motion blur)
> 
> Can you do a comparison with only splatfacto and compare with dn_splatter? Does dn_splatter do worse than splatfacto? You can get eval metrics using ns-eval command to see PSNR.

Thank you very much for your prompt reply。
The training effect of splatfacto is as follows，There will also be many floating objects like fog。
Also, if I want to capture data similar to the Replica dataset, what equipment do I need to use for shooting。
<img width="915" alt="1728553456631" src="https://github.com/user-attachments/assets/897b15d9-1bb9-4313-bd65-80d962da0eed">

**Comment by maturk** [comment]
Hi, I think maybe your "--normal-format" flag should be opengl, not opencv if using my scripts.

**Comment by cici19850** [comment]
> suggest

Thank you very much, your reply has been very helpful to me.
I will try using the advice you provided, thank you very much again.

**Comment by cici19850** [comment]
> Hi, I think maybe your "--normal-format" flag should be opengl, not opencv if using my scripts.
> 
> For debugging, I suggest first just enabling depth loss, and see if the PearsonDepth is improving the white walls or not. Then enable normal loss, and see if that issue is resolved.
> 
> Replica dataset is a "synthetic" dataset with ground truth depth, normal, and mesh data. It is unlikely that you will be able to make a similar quality dataset unless you have access to high precision laser scanners and your poses are millimiter good. If using a smartphone for capturing, make sure your images have little motion blur.

Thank you very much, your reply has been very helpful to me.
I will try using the advice you provided, thank you very much again.

**Comment by cici19850** [comment]
> Hi, I think maybe your "--normal-format" flag should be opengl, not opencv if using my scripts.
> 
> For debugging, I suggest first just enabling depth loss, and see if the PearsonDepth is improving the white walls or not. Then enable normal loss, and see if that issue is resolved.
> 
> Replica dataset is a "synthetic" dataset with ground truth depth, normal, and mesh data. It is unlikely that you will be able to make a similar quality dataset unless you have access to high precision laser scanners and your poses are millimiter good. If using a smartphone for capturing, make sure your images have little motion blur.

There is another issue, if using a smartphone for shooting and ensuring that the image has almost no motion blur.
Is it correct to execute the following command sequence：
1、python dn_splatter/scripts/convert_colmap.py 
2、python dn_splatter/scripts/normals_from_pretrain.py
3、python dn_splatter/scripts/align_depth.py
4、ns-train dn-splatter

**Comment by cici19850** [comment]
> Hi, I think maybe your "--normal-format" flag should be opengl, not opencv if using my scripts.
> 
> For debugging, I suggest first just enabling depth loss, and see if the PearsonDepth is improving the white walls or not. Then enable normal loss, and see if that issue is resolved.
> 
> Replica dataset is a "synthetic" dataset with ground truth depth, normal, and mesh data. It is unlikely that you will be able to make a similar quality dataset unless you have access to high precision laser scanners and your poses are millimiter good. If using a smartphone for capturing, make sure your images have little motion blur.

when I excute align_depth.py.Reminder information"Average depth alignment error for batch depths ",What is the reason for this？
<img width="638" alt="1728704071379" src="https://github.com/user-attachments/assets/55448913-5f4b-4285-96fc-6a541673afbb">

**Comment by cici19850** [comment]
> Hi, I think maybe your "--normal-format" flag should be opengl, not opencv if using my scripts.
> 
> For debugging, I suggest first just enabling depth loss, and see if the PearsonDepth is improving the white walls or not. Then enable normal loss, and see if that issue is resolved.
> 
> Replica dataset is a "synthetic" dataset with ground truth depth, normal, and mesh data. It is unlikely that you will be able to make a similar quality dataset unless you have access to high precision laser scanners and your poses are millimiter good. If using a smartphone for capturing, make sure your images have little motion blur.

Will the following prompt have an impact?
<img width="639" alt="1728708099006" src="https://github.com/user-attachments/assets/7f672f30-41d3-4a00-a770-c3ed285020ce">

**Comment by maturk** [comment]
@cici19850, the four commands you have ran are fine. You can skip step 1) if you use e.g. `ns-process-data` or some other tool to process your camera poses. 2) just gives you normal estimates and 3) converts COLMAP dataset SfM points to scale aligned mono-depth estimates. For more information about 3) I suggest looking at this [paper](https://arxiv.org/abs/2311.13398) which does the same thing (they use gradient descent to solve for scale and shift, but instead in my script I use the closed form solution). The "average depth alignment error for batch depths is..." warning relates to this step. It calculates how much the SfM points disagree with the monocular depth estimates. 

You can also skip step 3) and only run the `python dn_splatter/scripts/depth_from_pretrain.py` command which only generates monocular depth estimates (using Zoe) and skips the Colmap alignment step. This is okay if you are using the PearsonDepth loss, which is a relative loss. For other loss functions, the scale alignment is necessary. 

From my experience and experiments monocular depth supervision, even using PearsonDepth loss, does not perform as well as using e.g. iPhone ToF lidar data. Please see table below:
